certifier_service/certlib/certlib_policy.go

```diff
@@ -15,201 +15,73 @@
+	return ps
 }
 
-bool certifier::framework::policy_store::update_or_insert(const string &tag,
-                                                          const string &type,
-                                                          const string &value) {
-  int ent = find_entry(tag, type);
-  if (ent < 0) {
-    return add_entry(tag, type, value);
-  }
-  if (!put(ent, value))
-    return false;
-  return true;
+func PrintPolicyStore(ps *certprotos.PolicyStoreMessage) {
+	fmt.Printf("Maximum Entries: %d, current entries: %d\n", int(ps.GetMaxEnts()), len(ps.Entries))
+	for i := 0; i < len(ps.Entries); i++ {
+		fmt.Printf("   ")
+		PrintPolicyStoreEntry(ps.Entries[i])
+	}
 }
 
-bool certifier::framework::policy_store::delete_entry(unsigned ent) {
-  if (ent >= num_ents_)
-    return false;
-
-  delete entry_[ent];
-  for (unsigned i = ent; i < num_ents_; i++) {
-    entry_[i] = entry_[i + 1];
-  }
-  entry_[num_ents_ - 1] = nullptr;
-  num_ents_--;
-  return true;
+func FindPolicyStoreEntry(ps *certprotos.PolicyStoreMessage, tag string, valueType string) int {
+	for i := 0; i < len(ps.Entries); i++ {
+		if ps.Entries[i].GetTag() == tag && ps.Entries[i].GetType() == valueType {
+			return i
+		}
+	}
+	return -1
 }
 
-void certifier::framework::policy_store::print() {
-  printf("Number of entries: %d, max number of ents: %d\n",
-         num_ents_,
-         max_num_ents_);
-
-  for (unsigned i = 0; i < num_ents_; i++) {
-    printf("  Entry %3d: ", i);
-    entry_[i]->print();
-    printf("\n");
-  }
-}
-
-ool certifier::framework::policy_store::Serialize(string *psout) {
-  policy_store_message psm;
-
-  psm.set_max_ents(max_num_ents_);
-
-  for (unsigned i = 0; i < num_ents_; i++) {
-    policy_store_entry *pe = psm.add_entries();
-    store_entry *       se = entry_[i];
-    pe->set_tag(se->tag_);
-    pe->set_type(se->type_);
-    pe->set_value(se->value_);
-  }
-
-  return (psm.SerializeToString(psout));
+func InsertOrUpdatePolicyStoreEntry(ps *certprotos.PolicyStoreMessage, tag string, valueType string, value []byte) bool {
+	ent := FindPolicyStoreEntry(ps, tag, valueType)
+	if ent >= 0 {
+		ps.Entries[ent].Value = value
+		return true
+	}
+	if len(ps.Entries) >= int(ps.GetMaxEnts()) {
+		return false
+	}
+	pse := NewPolicyStoreEntry(tag, valueType, value)
+	ps.Entries = append(ps.Entries, pse)
+	return true
 }
 
-bool certifier::framework::policy_store::Deserialize(string &in) {
-
-  policy_store_message psm;
-
-  if (!psm.ParseFromString(in))
-    return false;
-
-  if (psm.has_max_ents()) {
-    max_num_ents_ = psm.max_ents();
-  } else {
-    max_num_ents_ = MAX_NUM_ENTRIES;
-  }
-
-  for (int i = 0; i < psm.entries_size(); i++) {
-    const policy_store_entry &pe = psm.entries(i);
-    store_entry *             se = new store_entry();
-    entry_[i] = se;
-    se->tag_ = pe.tag();
-    se->type_ = pe.type();
-    se->value_ = pe.value();
-  }
-  num_ents_ = psm.entries_size();
-
-  return true;
+func PolicyStoreDeleteEntry(ent int) {
```

Comment:
Looks like this is never implemented and used?

Reply:
It is in simpleserver.go lines 805,838